src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/SplitNCigarReads.java

```diff
@@ -134,32 +127,48 @@ public void closeTool() {
 
 
     /**
-     * Goes through the cigar string of the read and create new reads for each consecutive non-N elements (while hard clipping the rest of the read).
-     * For example: for a read with cigar '1H2M2D1M2N1M2I1N1M2S' 3 new reads will be created with cigar strings: 1H2M2D1M, 1M2I and 1M2S
+     * Goes through the cigar string of the read and create new reads for each consecutive non-N elements (while soft clipping the rest of the read) that are supplemental to eachother.
+     * For example: for a read with cigar '1H2M2D1M2N1M2I1N1M2S' 3 new reads will be created with cigar strings: 1H2M2D1M6S, 1H3S1M2I2S and 1H6S1M1S
      *
      * @param read     the read to split
      */
     public static GATKRead splitNCigarRead(final GATKRead read, OverhangFixingManager manager) {
         final int numCigarElements = read.numCigarElements();
+        List<GATKRead> splitReads = new LinkedList<>();
+        boolean sectionHasMatch = false;
 
         int firstCigarIndex = 0;
         for ( int i = 0; i < numCigarElements; i++ ) {
             final CigarElement cigarElement = read.getCigar().getCigarElement(i);
-            if (cigarElement.getOperator() == CigarOperator.N) {
-                manager.addRead(splitReadBasedOnCigar(read, firstCigarIndex, i, manager));
+            CigarOperator op = cigarElement.getOperator();
+            if (op == CigarOperator.M || op == CigarOperator.EQ || op == CigarOperator.X ||
+                    op == CigarOperator.I || op == CigarOperator.D) {
+                sectionHasMatch = true;
+            }
+
+            if (op == CigarOperator.N) {
+                if (sectionHasMatch) {
+                    splitReads.add(splitReadBasedOnCigar(read, firstCigarIndex, i, null));
+                }
                 firstCigarIndex = i+1;
+                sectionHasMatch = false;
             }
         }
 
         // if there are no N's in the read
-        if (firstCigarIndex == 0) {
+        if (splitReads.size() < 1) {
             manager.addRead(read);
+            return read;
         }
         //add the last section of the read: from the last N to the the end of the read
         // (it will be done for all the usual cigar string that does not end with N)
-        else if (firstCigarIndex < numCigarElements) {
-            manager.addRead(splitReadBasedOnCigar(read, firstCigarIndex, numCigarElements, null));
+        else if ((firstCigarIndex < numCigarElements) && sectionHasMatch) {
+            splitReads.add(splitReadBasedOnCigar(read, firstCigarIndex, numCigarElements, null));
```

Comment:
It looks like both times you call splitReadBasedOnCigar you pass null in as the manager. Can you remove it from the function itself? That way you can delete lines 205-210 (since it's never touched).